src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationExtensions.cs

```diff
@@ -0,0 +1,90 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using Dapr.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.DaprSecretStore;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Extension methods for registering <see cref="DaprSecretStoreConfigurationProvider"/> with <see cref="IConfigurationBuilder"/>.
+    /// </summary>
+    public static class DaprSecretStoreConfigurationExtensions
+    {
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Dapr Secret Store.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="store">Dapr secret store name.</param>
+        /// <param name="secrets">The secrets to retrieve.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddDaprSecretStore(
+            this IConfigurationBuilder configurationBuilder,
+            string store,
+            DaprSecretDescriptor[] secrets)
```

Comment:
Why are these required and why is it an array instead of an IEnumerable or IList or something like that.

Reply:
Hi @davidfowl just gave a partiall answer here: https://github.com/dapr/dotnet-sdk/pull/238#discussion_r390827188

Reply:
secrets parameter is now and IEnumerbale<DaprSecretDescriptor> (04cc749)